packages/core/src/extension/subscribers/subscribe-shipping-country-change.ts

```diff
@@ -0,0 +1,45 @@
+import { InternalCheckoutSelectors, ReadableCheckoutStore } from '../../checkout';
+import { ExtensionEventType } from '../extension-client';
+import { ExtensionEventBroadcaster } from '../extension-event-broadcaster';
+
+import { ExtensionChangeSubscriber } from './extension-change-subscriber';
+
+export const subscribeShippingCountryChange: ExtensionChangeSubscriber = (
+    store: ReadableCheckoutStore,
+    broadcaster: ExtensionEventBroadcaster,
+) => {
+    let countryCodes: Record<string, string> = getShippingCountryCodes(store.getState());
+
+    return store.subscribe(
```

Comment:
This looks good, but as I see we need to have a consignment before sending the event for country change? If that's the case we might need to communicate accordingly 🤔

Reply:
Hrm, I thought we want to broadcast changes that have been committed, not uncommitted changes. If we're looking to broadcaster the latter, we have to change the approach where we listen for UI events instead.